fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/LaoCreateFragment.java

```diff
@@ -141,24 +131,50 @@ private void setupCreateButton() {
               Objects.requireNonNull(binding.serverUrlEntryEditText.getText()).toString();
           String laoName =
               Objects.requireNonNull(binding.laoNameEntryEditText.getText()).toString();
+          boolean isWitnessingEnabled =
+              Boolean.TRUE.equals(viewModel.isWitnessingEnabled().getValue());
           Timber.tag(TAG).d("creating lao with name %s", laoName);
           List<PublicKey> witnesses = witnessingViewModel.getScannedWitnesses();
 
           networkManager.connect(serverAddress);
           requireActivity()
               .startActivity(
                   ConnectingActivity.newIntentForCreatingDetail(
-                      requireContext(), laoName, witnesses));
+                      requireContext(), laoName, witnesses, isWitnessingEnabled));
         });
   }
 
-  private void setupCancelButton() {
-    binding.buttonCancelLaunch.setOnClickListener(
+  private void setupClearButton() {
+    binding.buttonClearLaunch.setOnClickListener(
         v -> {
           Objects.requireNonNull(binding.laoNameEntryEditText.getText()).clear();
-          HomeActivity.setCurrentFragment(
-              getParentFragmentManager(), R.id.fragment_home, HomeFragment::newInstance);
+          Objects.requireNonNull(binding.serverUrlEntryEditText.getText()).clear();
+          binding.enableWitnessingSwitch.setChecked(false);
+          witnessingViewModel.setWitnesses(Collections.emptyList());
+        });
+  }
+
+  private void setupWitnessingSwitch() {
+    binding.enableWitnessingSwitch.setOnCheckedChangeListener(
+        (button, isChecked) -> {
+          viewModel.setIsWitnessingEnabled(isChecked);
+          if (isChecked) {
+            binding.addWitnessButton.setVisibility(View.VISIBLE);
+            if (!witnessingViewModel.getScannedWitnesses().isEmpty()) {
+              binding.witnessesTitle.setVisibility(View.VISIBLE);
+              binding.witnessesList.setVisibility(View.VISIBLE);
+            }
+            binding.enableWitnessingSwitch.setText(R.string.lao_create_disable_witnessing_switch);
```

Comment:
I think the labeling of the witnessing switch is a bit confusing, and it could just say _Enable witnessing protocol_ in both cases. This way I thought the protocol was disabled when the switch is checked.

Reply:
Agreed